File: code-coaches/medium/camel-to-snake/camel_to_snake.py
def to_snake_case(camel_case_string: str) -> str:
    snake_case_string = ""

    for i, char in enumerate(camel_case_string):
        if i != 0:
            previous_char = camel_case_string[i - 1]
            if (
                char.upper() == char
                and previous_char.isalpha()
                or (char.isalpha() and previous_char.isdigit())
            ):
                snake_case_string += "_"

        snake_case_string += char.lower()

    return snake_case_string


def main():
    camel_case_string = input()
    print(to_snake_case(camel_case_string))


if __name__ == "__main__":
    main()

File: code-coaches/medium/camel-to-snake/camel_to_snake.cpp
#include <iostream>
#include <string>

std::string to_snake_case(std::string camel_case_string)
{
    std::string snake_case_string = "";

    for (int i = 0; i < camel_case_string.length(); i++)
    {
        char current_char = camel_case_string[i];

        if (i != 0)
        {
            char previous_char = camel_case_string[i - 1];

            if ((isupper(current_char) && isalpha(previous_char))
                || (isalpha(current_char) && isdigit(previous_char)))
            {
                snake_case_string += "_";
            }
        }

        snake_case_string += tolower(current_char);
    }

    return snake_case_string;
}

int main()
{
    std::string camel_case_string;
    getline(std::cin, camel_case_string);

    std::cout << to_snake_case(camel_case_string);

    return 0;
}
